src/FEValues/face_integrals.jl

```diff
@@ -0,0 +1,166 @@
+##################
+# All 1D RefCube #
+##################
+function create_face_quad_rule{T, shape <: RefCube}(quad_rule::QuadratureRule{0, shape, T}, ::Interpolation{1, shape})
+    w = getweights(quad_rule)
+    face_quad_rule = QuadratureRule{1, shape, T}[]
+
+    # Face 1
+    new_points = [Vec{1, T}((-one(T),))] # ξ = -1
+    push!(face_quad_rule, QuadratureRule{1, shape, T}(w, new_points))
+    # Face 2
+    new_points = [Vec{1, T}((one(T),))] # ξ = 1
+    push!(face_quad_rule, QuadratureRule{1, shape, T}(w, new_points))
+
+    return face_quad_rule
+end
+
+function weighted_normal{T}(::Tensor{2, 1, T}, ::FaceValues{1, T, RefCube}, face::Int)
+    face == 1 && return Vec{1,T}((-one(T),))
+    face == 2 && return Vec{1,T}((one(T),))
+    error("unknown face number: $face")
+end
+
+##################
+# All 2D RefCube #
+##################
+function create_face_quad_rule{T, shape <: RefCube}(quad_rule::QuadratureRule{1, shape, T}, ::Interpolation{2, shape})
+    w = getweights(quad_rule)
+    p = getpoints(quad_rule)
+    n_points = length(w)
+    face_quad_rule = QuadratureRule{2, shape, T}[]
+
+    # Face 1
+    new_points = [Vec{2, T}((p[i][1], -one(T))) for i in 1:n_points] # ξ = t, η = -1
+    push!(face_quad_rule, QuadratureRule{2, shape, T}(w, new_points))
+    # Face 2
+    new_points = [Vec{2, T}((one(T), p[i][1])) for i in 1:n_points] # ξ = 1, η = t
+    push!(face_quad_rule, QuadratureRule{2, shape, T}(w, new_points))
+    # Face 3
+    new_points = [Vec{2, T}((p[i][1], one(T))) for i in 1:n_points] # ξ = t, η = 1
+    push!(face_quad_rule, QuadratureRule{2, shape, T}(w, new_points))
+    # Face 4
+    new_points = [Vec{2, T}((-one(T), p[i][1])) for i in 1:n_points] # ξ = -1, η = t
+    push!(face_quad_rule, QuadratureRule{2, shape, T}(w, new_points))
+
+    return face_quad_rule
+end
+
+function weighted_normal{T}(J::Tensor{2, 2}, ::FaceValues{2, T, RefCube}, face::Int)
+    @inbounds begin
+        face == 1 && return Vec{2}(( J[2,1], -J[1,1]))
```

Comment:
Sweet!
I wonder how this will work out if we would have quadratic geometric interpolation. I am pretty sure we would need a completely other strategy?